tests/portal-chooser/portal-chooser-test.sh

```
@@ -76,87 +147,111 @@ call1() { head -1 "$WORK/$1/portal-calls.jsonl" 2>/dev/null; }
+if session_ok noportal "$rc"; then
+  grep -q "portal service(s) omitted" "$WORK/noportal.log" &&
+    ok "the bus could activate everything except the portal" ||
+    bad "the no-portal bus was not built as intended"
+  # A PIN on measured behaviour: GTK does NOT fall back to the native chooser, so no
+  # extra window appears. That stays true after the #510 fix and is still worth
+  # holding -- the remedy is a message rendered by the WebView, not a second window,
+  # and the day a window does appear here somebody has reintroduced the in-process
+  # chooser that corrupts the GLib heap. If this fails, read it before "fixing" it.
+  wins=$(count_lines "$WORK/noportal/windows-after.txt")
+  wins_ref=$(count_lines "$WORK/cancel/windows-after.txt")
+  [ "$wins" = "$wins_ref" ] &&
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)